From 82e225e09ea111e00636c630e790483d4a459690 Mon Sep 17 00:00:00 2001
From: zhongfly <11155705+zhongfly@users.noreply.github.com>
Date: Sat, 12 Sep 2026 21:12:24 +0800
Subject: [PATCH] llvm: update to release/23.x branch

Disable newly introduced LLVM and Clang tools that are not required by the toolchain. Remove obsolete LLVM 23 CMake options.
---
 toolchain/llvm/llvm.cmake | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/toolchain/llvm/llvm.cmake b/toolchain/llvm/llvm.cmake
index 2c9a972..7850abb 100644
--- a/toolchain/llvm/llvm.cmake
+++ b/toolchain/llvm/llvm.cmake
@@ -1,4 +1,4 @@
-set(clang_version "22")
+set(clang_version "23")
 ExternalProject_Add(llvm
     GIT_REPOSITORY https://github.com/llvm/llvm-project.git
     SOURCE_DIR ${SOURCE_LOCATION}
@@ -6,7 +6,7 @@ ExternalProject_Add(llvm
     GIT_CLONE_POST_COMMAND "sparse-checkout set --no-cone /* !*/test !/lldb !/mlir !/clang-tools-extra !/mlir !/polly !/bolt !/flang"
     UPDATE_COMMAND ""
     GIT_REMOTE_NAME origin
-    GIT_TAG release/22.x
+    GIT_TAG release/23.x
     LIST_SEPARATOR ,
     CONFIGURE_COMMAND ${EXEC} CONF=1 PATH=$O_PATH cmake -H<SOURCE_DIR>/llvm -B<BINARY_DIR>
         -G Ninja
@@ -30,9 +30,7 @@ ExternalProject_Add(llvm
         -DLLVM_ENABLE_LIBXML2=OFF
         -DLLVM_ENABLE_LIBEDIT=OFF
         -DLLVM_ENABLE_LIBPFM=OFF
-        -DLLVM_ENABLE_TERMINFO=OFF
         -DLLVM_ENABLE_LTO=${LLVM_ENABLE_LTO}
-        -DCLANG_ENABLE_ARCMT=OFF
         -DCLANG_ENABLE_STATIC_ANALYZER=OFF
         -DLLVM_INCLUDE_TESTS=OFF
         -DLLVM_INCLUDE_EXAMPLES=OFF
@@ -80,14 +78,15 @@ ExternalProject_Add(llvm
         -DCLANG_TOOL_CLANG_REPL_BUILD=OFF
         -DCLANG_TOOL_CLANG_SCAN_DEPS_BUILD=ON
         -DCLANG_TOOL_CLANG_SHLIB_BUILD=OFF
+        -DCLANG_TOOL_CLANG_SSAF_ANALYZER_BUILD=OFF
+        -DCLANG_TOOL_CLANG_SSAF_FORMAT_BUILD=OFF
+        -DCLANG_TOOL_CLANG_SSAF_LINKER_BUILD=OFF
         -DCLANG_TOOL_DIAGTOOL_BUILD=OFF
         -DCLANG_TOOL_LIBCLANG_BUILD=OFF
         -DCLANG_TOOL_NVPTX_ARCH_BUILD=OFF
         -DCLANG_TOOL_SCAN_BUILD_BUILD=OFF
         -DCLANG_TOOL_SCAN_BUILD_PY_BUILD=OFF
         -DCLANG_TOOL_SCAN_VIEW_BUILD=OFF
-        -DLLVM_TOOL_BUGPOINT_BUILD=OFF
-        -DLLVM_TOOL_BUGPOINT_PASSES_BUILD=OFF
         -DLLVM_TOOL_DSYMUTIL_BUILD=OFF
         -DLLVM_TOOL_DXIL_DIS_BUILD=OFF
         -DLLVM_TOOL_GOLD_BUILD=OFF
@@ -98,6 +97,7 @@ ExternalProject_Add(llvm
         -DLLVM_TOOL_LLVM_AS_FUZZER_BUILD=OFF
         -DLLVM_TOOL_LLVM_BCANALYZER_BUILD=OFF
         -DLLVM_TOOL_LLVM_C_TEST_BUILD=OFF
+        -DLLVM_TOOL_LLVM_CAS_FUZZER_BUILD=OFF
         -DLLVM_TOOL_LLVM_CAT_BUILD=OFF
         -DLLVM_TOOL_LLVM_CFI_VERIFY_BUILD=OFF
         -DLLVM_TOOL_LLVM_CGDATA_BUILD=OFF
@@ -132,6 +132,7 @@ ExternalProject_Add(llvm
         -DLLVM_TOOL_LLVM_LIPO_BUILD=OFF
         -DLLVM_TOOL_LLVM_LTO_BUILD=OFF
         -DLLVM_TOOL_LLVM_LTO2_BUILD=OFF
+        -DLLVM_TOOL_LLUBI_BUILD=OFF
         -DLLVM_TOOL_LLVM_MC_ASSEMBLE_FUZZER_BUILD=OFF
         -DLLVM_TOOL_LLVM_MC_BUILD=OFF
         -DLLVM_TOOL_LLVM_MC_DISASSEMBLE_FUZZER_BUILD=OFF
-- 
2.54.0.windows.1

